02_classify_mutations/batch_effects/bc_titration.ipynb

```diff
@@ -0,0 +1,528 @@
+    "        ax.set_ylim(-0.1, 1.1)\n",
+    "        ax.set_ylabel('AUPR')\n",
+    "        ax.set_title('{} titration performance'.format(gene))\n",
+    "        \n",
+    "        # plot signal and shuffled baseline, linear and nonlinear\n",
+    "        import itertools as it\n",
+    "        for signal, model in it.product(\n",
+    "            ('signal', 'shuffled'),\n",
+    "            ('linear', 'nonlinear')\n",
+    "        ):\n",
+    "            baseline_vals = baseline_df[\n",
+    "                (baseline_df.identifier == gene) &\n",
+    "                (baseline_df.data_type == 'test') &\n",
+    "                (baseline_df.model == model) &\n",
+    "                (baseline_df.signal == signal)\n",
+    "            ]['aupr'].values\n",
+    "\n",
+    "            # get mean and plot as dotted line\n",
+    "            baseline_mean = np.mean(baseline_vals)\n",
+    "            color_ix = 1 if model == 'nonlinear' else 0\n",
+    "            style = '--' if signal == 'signal' else ':'\n",
+    "            ax.axhline(y=baseline_mean, linestyle=style, linewidth=3,\n",
+    "                       color=sns.color_palette()[color_ix])\n",
+    "            print('{}, {} mean: {:.3f}'.format(model, signal, baseline_mean))\n",
+    "\n",
+    "            # get bootstrapped 95% CI, using seaborn method\n",
+    "            # plot as transparent shaded region\n",
+    "            baseline_ci = sns.utils.ci(\n",
+    "                sns.algorithms.bootstrap(baseline_vals,\n",
+    "                                         func=np.mean,\n",
+    "                                         n_boot=1000,\n",
+    "                                         units=None,\n",
+    "                                         seed=cfg.default_seed)\n",
+    "            )\n",
+    "            ax.axhspan(baseline_ci[0], baseline_ci[1], alpha=0.3,\n",
+    "                       facecolor=sns.color_palette()[color_ix])\n",
+    "            \n",
+    "            if ix == 0:\n",
+    "                from matplotlib.lines import Line2D\n",
+    "                custom_legend_lines = [\n",
+    "                    Line2D([0], [0], color='black', linestyle='--'),\n",
+    "                    Line2D([0], [0], color='black', linestyle=':')\n",
+    "                ]\n",
+    "                handles, labels = ax.get_legend_handles_labels()\n",
+    "                l = ax.legend(\n",
+    "                    [Line2D([0], [0], color='black', linestyle='--')] + handles +[Line2D([0], [0], color='black', linestyle='--')] + custom_legend_lines,\n",
+    "                    ['model'] + ['linear', 'nonlinear'] + ['signal'] + ['signal', 'shuffled']\n",
+    "                , ncol=2)\n",
+    "                l.legendHandles[0].set_visible(False)\n",
+    "                l.legendHandles[3].set_visible(False)\n",
+    "            else:\n",
+    "                ax.legend().set_visible(False)\n",
```

Comment:
It's interesting to me that different problems take different numbers of features to reach 100 percent accuracy. I wonder whether that's a function of the process (some features are more corrupted by be correction) or the data (some problems have something that makes it hard to learn useful signal). If you're looking for a positive control to run, you could titrate in setting X number of features to the true labels and see how long it takes the model to hit 100% accuracy

Reply:
This is an interesting idea! My guess is that the positive control would hit 100% accuracy pretty quickly but it would be good to try.